Update rate must be an integer?

**Describe the bug**
In a simple controller package using `generate_parameter_library`, using a floating-point number in an `update_rate.yaml` file throws an error that the update rate param must be an integer.

```
(humble) dan@computer:~/ros/minimal_ws$ ros2 launch dz_minimal_controller controller_load.launch.py
[INFO] [launch]: All log files can be found below /home/dan/.ros/log/2022-11-05-12-08-27-470583-computer-191619
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [ros2_control_node-1]: process started with pid [191620]
[INFO] [spawner-2]: process started with pid [191622]
[ros2_control_node-1] terminate called after throwing an instance of 'rclcpp::exceptions::InvalidParameterTypeException'
[ros2_control_node-1]   what():  parameter 'update_rate' has invalid type: expected [integer] got [double]
[spawner-2] [INFO] [1667664507.716583768] [spawner_dz_controller]: Waiting for '/controller_manager' node to exist
[ERROR] [ros2_control_node-1]: process has died [pid 191620, exit code -6, cmd '/home/dan/.local/opt/mambaforge/envs/humble/lib/controller_manager/ros2_control_node --ros-args --params-file /tmp/launch_params_pkemfwt8 --params-file /home/dan/ros/minimal_ws/install/share/dz_minimal_controller/config/update_rate.yaml --params-file /home/dan/ros/minimal_ws/install/share/dz_minimal_controller/config/controller_params.yaml'].
[spawner-2] [INFO] [1667664509.729789056] [spawner_dz_controller]: Waiting for '/controller_manager' node to exist
```

**To Reproduce**
Steps to reproduce the behavior:
 * Possibly can be triggered in other packages by finding the update rate YAML file and changing the rate to a floating-point value
 * Otherwise, the `dz/fp-update-rate` branch here reproduces it with a simple controller: https://github.com/danzimmerman/dz_minimal_controller/tree/dz/fp-update-rate

**Expected behavior**
The update rate in Hertz can take non-integer values per the documentation:

https://control.ros.org/master/doc/ros2_control/controller_manager/doc/userdoc.html#parameters

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04 Focal
 - Version: Humble, installed via [Robostack](https://robostack.github.io/)
- Anything that may be unusual about your environment:
 - Mixed 2.12 and 2.13 release binaries from Robostack

**Additional context**

Possibly this is user error on my package level. Is there something about defaults in [`generate_parameter_library`](https://github.com/PickNikRobotics/generate_parameter_library/)? 

Maybe this is more of an issue for that repo? I've been working to try to use `ros2_control` properly with `generate_parameter_library` by looking at Joint Trajectory Controller and Admittance Controller, maybe there's something I'm missing.

--- a/controller_manager/doc/userdoc.rst
+++ b/controller_manager/doc/userdoc.rst
@@ -57,7 +57,7 @@ robot_description (mandatory; string)
   String with the URDF string as robot description.
   This is usually result of the parsed description files by ``xacro`` command.
 
-update_rate (mandatory; double)
+update_rate (mandatory; integer)
   The frequency of controller manager's real-time update loop.
   This loop reads states from hardware, updates controller and writes commands to hardware.
 

